File: scripts/convert-images.mjs
/**
 * Convert downloaded Pexels originals to responsive WebP at multiple widths.
 * Run: node scripts/convert-images.mjs
 */
import sharp from "sharp";
import { readdir, mkdir } from "fs/promises";
import { join, basename, extname } from "path";

const SOURCE_DIR = "design-sources/originals";
const OUTPUT_DIR = "public/images/editorial";
const WIDTHS = [640, 960, 1280, 1600];
const QUALITY = 82;

async function convert() {
  await mkdir(OUTPUT_DIR, { recursive: true });
  const files = await readdir(SOURCE_DIR);
  const images = files.filter((f) => /\.(jpg|jpeg|png)$/i.test(f));

  for (const file of images) {
    const name = basename(file, extname(file));
    const src = join(SOURCE_DIR, file);
    const meta = await sharp(src).metadata();
    console.log(`\n${file}: ${meta.width}x${meta.height}`);

    for (const w of WIDTHS) {
      if (meta.width && w > meta.width) {
        console.log(`  skip ${w}w (source is ${meta.width}px)`);
        continue;
      }
      const out = join(OUTPUT_DIR, `${name}-${w}w.webp`);
      await sharp(src).resize(w).webp({ quality: QUALITY }).toFile(out);
      console.log(`  → ${out}`);
    }

    // Also create a default (largest that fits)
    const defaultW = Math.min(1280, meta.width || 1280);
    const defaultOut = join(OUTPUT_DIR, `${name}.webp`);
    await sharp(src).resize(defaultW).webp({ quality: QUALITY }).toFile(defaultOut);
    console.log(`  → ${defaultOut} (default)`);
  }

  console.log("\n✓ All images converted.");
}

convert().catch(console.error);

File: scripts/convert_images.py
import os
from PIL import Image

SOURCE_DIR = r"c:\Users\khare\Downloads\Solve-Sprint-main\Solve-Sprint-main\design-sources\originals"
OUTPUT_DIR = r"c:\Users\khare\Downloads\Solve-Sprint-main\Solve-Sprint-main\public\images\editorial"
WIDTHS = [640, 960, 1280, 1600]
QUALITY = 82

os.makedirs(OUTPUT_DIR, exist_ok=True)

files = [f for f in os.listdir(SOURCE_DIR) if f.lower().endswith(('.jpg', '.jpeg', '.png'))]

for file in files:
    name = os.path.splitext(file)[0]
    src_path = os.path.join(SOURCE_DIR, file)
    with Image.open(src_path) as img:
        orig_w, orig_h = img.size
        print(f"\n{file}: {orig_w}x{orig_h}")
        
        # Convert & save for each width
        for w in WIDTHS:
            if w > orig_w:
                print(f"  skip {w}w (source is {orig_w}px)")
                continue
            h = int(orig_h * (w / orig_w))
            resized = img.resize((w, h), Image.Resampling.LANCZOS)
            out_path = os.path.join(OUTPUT_DIR, f"{name}-{w}w.webp")
            resized.save(out_path, "WEBP", quality=QUALITY)
            print(f"   Saved {out_path}")
            
        # Default full file (max 1280 or original)
        default_w = min(1280, orig_w)
        default_h = int(orig_h * (default_w / orig_w))
        default_img = img.resize((default_w, default_h), Image.Resampling.LANCZOS)
        default_out = os.path.join(OUTPUT_DIR, f"{name}.webp")
        default_img.save(default_out, "WEBP", quality=QUALITY)
        print(f"   Saved default {default_out}")

print("\nConverted all images to WebP successfully.")
